### DISCUSSION TITLE
Decoding of Refresh Token/ Access Token

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16826] [Archive]
1) How do I decode the refresh_token that is passed as a cookie to access a particular User from database ' directus_users '?
    I want functionality which is same as when clicked on profile page of directus which shos the current user that is logged in.

2) How do I use access_token that is generated when user signs in to access an API??
    I am stuck at these from  2 days. Please help.

3) How do I get User id with the help of access_token or if any other method to get user id..?

**Comment by raphourbe** [comment]
Hi,

You can use the `directus-sdk` (see [doc](https://docs.directus.io/reference/sdk.html#installation)) to handle the authentication part and retrieve the user's data, without having to deal with the access_token and refresh_token. To authenticate, you have two example here: https://docs.directus.io/reference/sdk.html#login

Then you will have access to the user `await directus.users.me.read();` as mentioned on the User part of the same doc.

If you do not want to use the directus-sdk, then you should handle authentication yourself. Regarding what it is used for and the technology you use, it may vary.

But once you have your `access_token`, you can access the API using `fetch` requests and a header.

```js
const headers = {
  "Content-Type": "application/json",
  Authorization: `Bearer ${access_token}`,
};

const res = await fetch("directus-url.com/", {
  method: "POST",
  headers,
  body: JSON.stringify({
    your_query,
    your_variables,
  }),
});

if (!res.ok) {
  const error = new Error("An error occurred while fetching the data.");
  // Attach extra info to the error object.
  error.info = await res.json();
  error.status = res.status;
  throw error;
}

const json = await res.json();

return json;
```
